jwst/extract_1d/ifu.py

```diff
@@ -50,7 +56,8 @@ def ifu_extract1d(input_model, refname, source_type):
         slitname = "ANY"
 
     extract_params = ifu_extract_parameters(refname, slitname, source_type)
-
+    if subtract_background is not None:
+        extract_params['subtract_background'] = subtract_background
```

Comment:
For IFU data, there already is a `subtract_background` key in `extract_params`, so I think the user should get an info message if the reference file specified that background subtraction should be done but the user specified that it should be turned off, or vice versa.  Also, background subtraction is currently only done for a point source, so if the user specified background subtraction for a target that is an extended, perhaps we really should subtract a background (i.e. maybe the target is only a little extended.

Reply:
@philhodge If background subtraction for an extended source is enabled how should `radius`, `inner_bkg` and `outer_bkg` be defined? Should they also be added as parameters?

Reply:
My understanding is that background subtraction should not be enabled for an extended source.  It would be possible to do so, however, using a reference image.  But parameters `radius`, `inner_bkg`, and `outer_bkg` should not be defined for IFU data.

But that's not what we were talking about; we want to be able to extract the background of an IFU image and ignore the source(s).  That is, the background is what we want as the result, i.e. to populate the FLUX column.  We can do that (or will be able to) by using a reference image.  After creating a segmentation map to isolate the sources, the non-source regions are the background.  So we would create an integer mask that is set to 0 for every pixel that has significant source flux, and set to 1 for all other pixels.  Using that as a reference image, extract_1d would give us the background flux for an IFU cube.